x-pack/plugins/alerting_builtins/server/alert_types/geo_threshold/es_query_builder.ts

```diff
@@ -0,0 +1,169 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ILegacyScopedClusterClient } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { Logger } from '../../types';
+import { GEO_THRESHOLD_ID } from './alert_type';
+
+export const OTHER_CATEGORY = 'other';
+// Consider dynamically obtaining from config?
+const MAX_QUERY_SIZE = 10000;
+
+export async function getShapesFilters(
+  boundaryIndexTitle: string,
+  boundaryType: string,
+  boundaryGeoField: string,
+  geoField: string,
+  callCluster: ILegacyScopedClusterClient['callAsCurrentUser'],
+  log: Logger,
+  alertId: string,
+  boundaryNameField?: string
+) {
+  const filters: Record<string, unknown> = {};
+  const shapesIdsNamesMap: Record<string, unknown> = {};
+  switch (boundaryType) {
+    case 'entireIndex':
+      // Get all shapes in index
+      const boundaryData: SearchResponse<Record<string, unknown>> = await callCluster('search', {
+        index: boundaryIndexTitle,
+        body: {
+          size: MAX_QUERY_SIZE,
+        },
+      });
+      boundaryData.hits.hits.forEach(({ _index, _id }) => {
+        filters[_id] = {
+          geo_shape: {
+            [geoField]: {
+              indexed_shape: {
+                index: _index,
+                id: _id,
+                path: boundaryGeoField,
+              },
+            },
+          },
+        };
+      });
+      if (boundaryNameField) {
+        boundaryData.hits.hits.forEach(
+          ({ _source, _id }: { _source: Record<string, unknown>; _id: string }) => {
+            shapesIdsNamesMap[_id] = _source[boundaryNameField];
+          }
+        );
+      }
+      break;
+    default:
+      log.info(`alert ${GEO_THRESHOLD_ID}:${alertId} Unsupported type: ${boundaryType}`);
+  }
+  return {
+    shapesFilters: filters,
+    shapesIdsNamesMap,
+  };
+}
+
+export async function executeEsQueryFactory(
+  {
+    entity,
+    index,
+    dateField,
+    boundaryGeoField,
+    geoField,
+    boundaryIndexTitle,
+    boundaryType,
+  }: {
+    entity: string;
+    index: string;
+    dateField: string;
+    boundaryGeoField: string;
+    geoField: string;
+    boundaryIndexTitle: string;
+    boundaryType: string;
+    boundaryNameField?: string;
+  },
+  { callCluster }: { callCluster: ILegacyScopedClusterClient['callAsCurrentUser'] },
+  log: Logger,
+  shapesFilters: Record<string, unknown>
+) {
+  return async (
+    gteDateTime: Date | null,
+    ltDateTime: Date | null
+  ): Promise<SearchResponse<unknown> | undefined> => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const esQuery: Record<string, any> = {
+      index,
+      body: {
+        size: MAX_QUERY_SIZE,
+        aggs: {
+          shapes: {
+            filters: {
+              other_bucket_key: OTHER_CATEGORY,
+              filters: shapesFilters,
+            },
+            aggs: {
+              entitySplit: {
+                terms: {
```

Comment:
Add size param here:

fwiw: It's tough to determine this up-front. This is because it is a nested agg, and the number of shapes-filters determines the amount of buckets you have available in the sub-terms agg. The top-hits also adds a bucket.

So I would try something like `MAX_BUCKET_NUMBER/(shapesFilters.length * 2)`

(* 2 because the top-hots creates a bucket for each term as well).

^ fwiw this is completely off the cuff and we should doulbe check with ES-team to know how to determine the agg-limits more exactly.

Reply:
FYI, this was the issue preventing triggered alerts from picking up the latest point from `other`. The reason it usually worked fine within shapes was that there was usually <= 10 unique entities in any given shape. Obviously in `other` there were far more.